numba/tests/npyufunc/test_vectorize_decor.py

```diff
@@ -28,88 +28,82 @@ def vector_add(a, b):
-               uint32(uint32, uint32),
-               float32(float32, float32),
-               float64(float64, float64)]
-        basic_ufunc = vectorize(sig, target=target)(vector_add)
-        np_ufunc = np.add
-
-        def test(ty):
-            data = np.linspace(0., 100., 500).astype(ty)
-            result = basic_ufunc(data, data)
-            gold = np_ufunc(data, data)
-            np.testing.assert_allclose(gold, result)
+class BaseVectorizeDecor(object):
 
-        test(np.double)
-        test(np.float32)
-        test(np.int32)
-        test(np.uint32)
+    __test__ = False
+    target = None
+    funcs = {
+        'func1': sinc,
+        'func2': scaled_sinc,
+        'func3': vector_add,
+    }
 
-
-class TestVectorizeDecor(BaseVectorizeDecor):
+    def _run_and_compare(self, func, sig, A, *args, **kwargs):
+        numba_func = vectorize(sig, target=self.target)(func)
+        numpy_func = np.vectorize(func)
+        result = numba_func(A, *args)
+        gold = numpy_func(A, *args)
+        np.testing.assert_allclose(result, gold, **kwargs)
 
     @tag('important')
-    def test_cpu_1(self):
-        self._test_template_1('cpu')
+    def test_1(self):
+        sig = ['float64(float64)', 'float32(float32)']
+        func = self.funcs['func1']
+        A = np.arange(100, dtype=np.float64)
+        self._run_and_compare(func, sig, A)
 
     @tag('important')
-    def test_parallel_1(self):
-        self._test_template_1('parallel')
-
+    def test_2(self):
+        sig = [float64(float64), float32(float32)]
+        func = self.funcs['func1']
+        A = np.arange(100, dtype=np.float64)
+        self._run_and_compare(func, sig, A)
+    
     @tag('important')
-    def test_cpu_2(self):
-        self._test_template_2('cpu')
+    def test_3(self):
+        sig = ['float64(float64, uint32)']
+        func = self.funcs['func2']
+        A = np.arange(100, dtype=np.float64)
+        scale = np.uint32(3)
+        self._run_and_compare(func, sig, A, scale, atol=1e-8)
 
     @tag('important')
-    def test_parallel_2(self):
-        self._test_template_2('parallel')
+    def test_4(self):
+        sig = [
+            int32(int32, int32),
+            uint32(uint32, uint32),
+            float32(float32, float32),
+            float64(float64, float64),
+        ]
+        func = self.funcs['func3']
+        A = np.arange(100, dtype=np.float64)
+        self._run_and_compare(func, sig, A, A)
+        A = A.astype(np.float32)
+        self._run_and_compare(func, sig, A, A)
+        A = A.astype(np.int32)
+        self._run_and_compare(func, sig, A, A)
+        A = A.astype(np.uint32)
+        self._run_and_compare(func, sig, A, A)
 
-    @tag('important')
-    def test_cpu_3(self):
-        self._test_template_3('cpu')
 
-    @tag('important')
-    def test_parallel_3(self):
-        self._test_template_3('parallel')
+class TestCPUVectorizeDecor(unittest.TestCase, BaseVectorizeDecor):
+    __test__ = True 
+    target = 'cpu'
```

Comment:
The reason for inheriting `TestCase` here rather than in the base class is that `py.test` will try to run anything inheriting from `TestCase` so to prevent pytest trying to run the base class it can't inherit from `TestCase`.  In this way the tests can be run by both nose an pytest
![image](https://cloud.githubusercontent.com/assets/881019/19370974/ce6e09ca-91f2-11e6-88b4-05f419a57490.png)